rest/route/task_generate.go

```diff
@@ -101,10 +102,19 @@ func (h *generatePollHandler) Run(ctx context.Context) gimlet.Responder {
 		return gimlet.MakeJSONInternalErrorResponder(err)
 	}
 	shouldExit := false
-	if len(jobErrs) > 0 { // exit early if we know the error will keep recurring
-		jobErr := errors.New(strings.Join(jobErrs, ", "))
-		shouldExit = db.IsDocumentLimit(jobErr)
+	// Exit early if we know the error will keep recurring.
+	for _, errString := range jobErrs {
```

Comment:
The benefit of the previous implementation was that we didn't need to loop; if we're now looping, we can break after we find an exit.

Reply:
it seems to me the benefit of saving an extra iteration or two is not worth the indirection of a `break`